cluster-provision/gocli/docker/docker.go

```diff
@@ -232,6 +234,15 @@ func NewCleanupHandler(cli *client.Client, cleanupChan chan error, errWriter io.
 						err := cli.ContainerRemove(ctx, c, types.ContainerRemoveOptions{Force: true})
 						if err != nil {
 							fmt.Fprintf(errWriter, "%v\n", err)
+							failedContainers = append(failedContainers, c)
```

Comment:
What about skipping remove for dnsmasq container here, and removing it at the end instead deleting it, failing to delete and handing it please ?

Reply:
> What about skipping remove for dnsmasq container here, and removing it at the end instead deleting it, failing to delete and handing it please ?

`createdContainers` just contains a list of container IDs as strings so its hard to identify which container ID maps to the dnsmasq container. 
> 
> Btw do we know that we enter this loop ? asking because forceClean is false, so it means err != nil right ? I wonder why err != nil if so, but this is other story. (we do see that we fail to delete the dnsmasq so we do enter some cleanup logic, I didn't find any other than this)

publish-kubevirtci runs the provision so it hits `NewCleanupHandler` here - https://github.com/kubevirt/kubevirtci/blob/d14be236cab7d2f219398611dd0a8ade8ce8b35f/cluster-provision/gocli/cmd/provision.go#L131
which has `forceClean` as true.

Reply:
> > What about skipping remove for dnsmasq container here, and removing it at the end instead deleting it, failing to delete and handing it please ?
> 
> `createdContainers` just contains a list of container IDs as strings so its hard to identify which container ID maps to the dnsmasq container.

you can do something like filterByPrefix, but whatever you prefer